File: AuthService/src/main/java/com/microservice/auth/service/AuthService.java
package com.microservice.auth.service;

import com.microservice.auth.models.User;
import com.microservice.auth.models.UserLogin;
import org.springframework.stereotype.Service;

@Service
public interface AuthService {
            public String doSignUp(User _newUser);
            public String doLogIn(String _userLogin) throws Exception;
}

File: FrontEndService/src/app/auth/auth.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Auth } from '../models/auth/Auth';
import { environment } from 'src/environments/environment';
import { User } from '../models/auth/User';

@Injectable({
  providedIn: 'root'
})
export class AuthService {

  constructor(private httpClient:HttpClient) { }

  doLogin(auth:Auth){
    return this.httpClient.post(environment.backendGateway+"auth/v1/authenticate",auth,{responseType:'text'});
  }

  doSignup(user:User){
    return this.httpClient.post(environment.backendGateway+"auth/v1/signup",user,{responseType:'text'});
  }
}
